README.md

```diff
@@ -94,6 +94,23 @@ emitter.send(new CloudEvent({ specversion: Version.V03, source, type }))
   }).catch(console.error);
 ```
 
+#### Cloud Events
+
+All created `CloudEvent` objects are read-only.  If you need to update a property or add a new extension to an existing cloud event object, you can use the `cloneWith` method.  This will return a new `CloudEvent` with any update or new properties.  For example:
+
+```js
+const {
+  CloudEvent,
+} = require("cloudevents-sdk");
+
+// Create a new CloudEvent
+const ce = new CloudEvent({.....});
+
+// Add a new extension to an existing CloudEvent
+ce = ce.cloneWith({extension: "Value"});
```

Comment:
- Cannot reassign a `const`
- This line is confusing . Why don't we just create a new `ce`?
  - Maybe `const ce2 = ce.cloneWith...`?

Reply:
> Cannot reassign a const

doh